HW2/P2/parallel_vector.pyx

```diff
@@ -105,11 +110,17 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    # to parallelize data movement.  Be sure to avoid deadlock, as well as
    # double-locking.
    ##########
+
+
    with nogil:
-       for r in range(repeat):
-           for idx in range(src.shape[0]):
+       for r in range (repeat):
+           for idx in prange(src.shape[0],num_threads=4, schedule=dynamic):
                if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
+                   acquire(&(locks[src[idx]/N]))
```

Comment:
I recall using static scheduling instead of dynamic scheduling. Did you compare the approaches? I might be mistaken, but I thought the pset instructions recommended static scheduling so was there a particular reason you chose dynamic scheduling?